src/Compilers/CSharp/Portable/BoundTree/BoundPattern_ConvertedType.cs

```diff
@@ -16,7 +18,7 @@ internal partial class BoundPattern
 
     internal partial class BoundConstantPattern
     {
-        public override TypeSymbol ConvertedType => Value.Type;
+        public override TypeSymbol ConvertedType => Value.Type ?? InputType;
```

Comment:
>public override TypeSymbol ConvertedType => Value.Type ?? InputType; [](start = 8, length = 68)

Is semantic change covered by a test?

Reply:
I don't think it can occur in a correct program, where this property is used.  But a ! would not be correct, because the Type field can be null (just not during lowering).

---
In reply to: [386576225](https://github.com/dotnet/roslyn/pull/42068#discussion_r386576225) [](ancestors = 386576225)